Xwt/Xwt.Backends/IButtonBackend.cs

```diff
@@ -36,6 +36,12 @@ public interface IButtonBackend: IWidgetBackend
 		void SetButtonStyle (ButtonStyle style);
 		void SetButtonType (ButtonType type);
 		void SetContent (string label, bool useMnemonic, ImageDescription image, ContentPosition position);
+
+		/// <summary>
+		/// Sets the formatted text display for this menu item.
+		/// </summary>
+		/// <param name="text">The text to display with additional markup for formatting <see cref="Xwt.FormattedText"/></param>
+		void SetFormattedText (FormattedText text);
```

Comment:
I don't think it would make much of a difference on the implementation of the backend either way. I think what would make it easier, was if all the labels were converted to FormattedText objects then applied. But you would need a way to determine of the FormattedText did not in fact contain any formatting so that you could short curcuit the label creation and styling for plain text.

Reply:
well, it'd let us avoid double-init here: https://github.com/mono/xwt/pull/959/files#diff-75a1dc4c07ad775d27eb0deadeee862fR255. Whether FormattedText contains formatting or not can be determined by `FormattedText.Attributes.Count`.

Reply:
So my reasoning for setting it up that way was that if you change the `Label` of the button, it calls `SetContent`, which recreates the `labelWidget` internally. So setting the `Markup` property it should do all the things that setting the `Label` does, plus applying the proper formatting. This is why there is a check to see if the `labelWidget` already exists in `SetFormattedContent` instead of just recreating it every time. In this way, `SetFormattedText` uses `SetContent` to configure everything properly, and then `SetFormattedContent` extends that to add the proper formatting.